1-server-side/msal-client-credential-secret/src/main/java/ClientCredentialGrant.java

```diff
@@ -62,7 +62,7 @@ public static void main(String args[]) throws Exception{
 
     private static void GetOrCreateApp(String clientId, String secret, String authority) throws MalformedURLException {
 
-        if (app!=null) {
+        if (app == null) {
```

Comment:
Oh snap, I wonder who introduced this :(

Reply:
Yeah, just noticed that when testing CIAM with other flows (the other places you added this method are correct though!)